src/directives/avatar.ts

```diff
@@ -37,6 +37,48 @@ export default [
                 receiver: '=eeeReceiver',
                 resolution: '=eeeResolution',
             },
+            link: function(scope, elem, attrs) {
+                scope.$watch(
+                    () => scope.ctrl.receiver,
+                    (newReceiver: threema.Receiver, oldReceiver: threema.Receiver) => {
+                        // Register for receiver changes. When something relevant changes, call the update function.
+                        // This prevents processing the avatar more often than necessary.
+
+                        if (!hasValue(newReceiver)) {
+                            // New receiver has no value
+                            return;
+                        }
+                        if (!hasValue(oldReceiver)) {
+                            // New receiver has value, old receiver doesn't
```

Comment:
That depends on the calling code :slightly_smiling_face: Since this is a directive, and the receiver is passed in through the "API", the receiver changes if the calling code changes it.

From the point of view in the directive, we don't know what triggers the change. It's just defensive programming.

Reply:
What I meant was: "What is triggered by this condition" but not "What triggers this condition". :)

Reply:
Ah. That's if the watch target (`scope.ctrl.receiver`) was changed from `undefined` to a value. But that's AngularJS basics, right?